test/unit/serializer/index.spec.ts

```diff
@@ -244,6 +305,78 @@ describe('index.ts', () => {
 			err = error;
 		}
 
+		expect(jsonDeserialize).toHaveBeenCalledTimes(1);
+		expect(gzipDeserialize).toHaveBeenCalledTimes(1);
 		expect(err).not.toBeUndefined();
 	});
+
+	it('should serialize enqueued when enqueue is used', async () => {
+		const target = new Serializer(new JsonStrategy(), { enqueue: 1 });
+		const call1 = jest.fn();
+		const call2 = jest.fn();
+		jest
+			.spyOn(target, 'serializeFactory' as any)
+			.mockImplementationOnce(
+				() => () => wait(10).then(call1).then(constant('result 1')),
+			)
+			.mockImplementationOnce(
+				() => () => wait(2).then(call2).then(constant('result 2')),
+			);
+
+		const result1 = target.serialize('data 1');
+		const result2 = await target.serialize('data 2');
+
+		expect(call1).toHaveBeenCalledBefore(call2);
+		expect(await result1).toBe('result 1');
+		expect(result2).toBe('result 2');
+	});
+
+	it('should serialize not enqueued when enqueue is used but the number of requests is lesser or equal than the pool size', async () => {
+		const target = new Serializer(new JsonStrategy(), { enqueue: 2 });
+		const call1 = jest.fn();
+		const call2 = jest.fn();
+		jest
+			.spyOn(target, 'serializeFactory' as any)
+			.mockImplementationOnce(
+				() => () => wait(10).then(call1).then(constant('result 1')),
+			)
+			.mockImplementationOnce(
+				() => () => wait(2).then(call2).then(constant('result 2')),
+			);
+
+		const result1 = target.serialize('data 1');
+		const result2 = await target.serialize('data 2');
+
+		expect(call2).toHaveBeenCalledBefore(call1);
+		expect(await result1).toBe('result 1');
+		expect(result2).toBe('result 2');
+	});
+
+	it('should serialize enqueued when enqueue is used and the number of requests is greater than the pool size', async () => {
+		const target = new Serializer(new JsonStrategy(), { enqueue: 2 });
+		const call1 = jest.fn();
+		const call2 = jest.fn();
+		const call3 = jest.fn();
+		jest
+			.spyOn(target, 'serializeFactory' as any)
+			.mockImplementationOnce(
+				() => () => wait(10).then(call1).then(constant('result 1')),
+			)
+			.mockImplementationOnce(
+				() => () => wait(4).then(call2).then(constant('result 2')),
+			)
+			.mockImplementationOnce(
+				() => () => wait(2).then(call3).then(constant('result 3')),
+			);
+
+		const result1 = target.serialize('data 1');
+		const result2 = target.serialize('data 2');
+		const result3 = await target.serialize('data 3');
+
+		expect(call2).toHaveBeenCalledBefore(call1);
+		expect(call1).toHaveBeenCalledBefore(call3);
```

Comment:
Call1 is called before call3 regardless of the time it takes, because the number of queues is lesser than the number of calls, and the calls are distributed between the queue using round robin